cmd/kubermatic-installer/cmd_mirror_images.go

```diff
@@ -372,9 +396,15 @@ func MirrorImagesFunc(logger *logrus.Logger, versions kubermaticversion.Versions
 				}
 			} else {
 				logger.WithField("registry", options.Registry).Info("🚀 Mirroring images…")
-				count, fullCount, err = images.CopyImages(ctx, logger, options.DryRun, sets.List(imageSet), options.Registry, userAgent)
+				count, fullCount, failedImages, err = images.CopyImages(ctx, logger, options.DryRun, sets.List(imageSet), options.Registry, userAgent)
 				if err != nil {
-					return fmt.Errorf("failed to mirror all images (successfully copied %d/%d): %w", count, fullCount, err)
+					// Format failed images into a clean list
+					errorMsg := fmt.Sprintf("Failed to mirror all images: %s\nFailed images:\n", err)
+					for _, img := range failedImages {
+						errorMsg += fmt.Sprintf("  - %s\n", img)
+					}
+					errorMsg += "Please check the logs for more details or retry the failed images"
+					return errors.New(errorMsg)
```

Comment:
You can move this formatting in the `copyImages` method instead and use https://github.com/kubermatic/kubermatic/blob/main/pkg/util/errors/aggregate.go#L35-L50 for the error aggregation.